File: Data Structures & Algorithms/is-anagram/submission-11.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashSet = {}
        for char in s:
            if char in hashSet:
                hashSet[char] += 1
            else:
                hashSet[char] = 1

        for char in t:
            if char not in hashSet or hashSet[char] == 0:
                return False
            else:
                hashSet[char] -= 1
            
        for key in hashSet:
            if hashSet[key] != 0:
                return False

        return True

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-11.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> inputSet = new HashSet<>();

        for(int num: nums){
            inputSet.add(num);
        }

        int max_value = 0; 

        for(int i = 0; i < nums.length; i++){
            if(inputSet.contains(nums[i] - 1)){
                continue;
            }

            int tempValue = nums[i];

            while(inputSet.contains(tempValue)){
                tempValue += 1;
            }

            max_value = Math.max(max_value, tempValue - nums[i]);
        }

        return max_value;
    }
}
